readthedocs/subscriptions/signals.py

```diff
@@ -34,23 +34,43 @@ def remove_stripe_subscription(sender, instance, using, **kwargs):
 
 # pylint: disable=unused-argument
 @receiver(post_save, sender=Organization)
-def update_billing_information(sender, instance, created, **kwargs):
-    """Update billing email information."""
+def update_stripe_customer(sender, instance, created, **kwargs):
+    """Update email and metadata attached to the stripe customer."""
     if created:
         return
 
     organization = instance
     log.bind(organization_slug=organization.slug)
-    # pylint: disable=broad-except
-    try:
-        s_customer = stripe.Customer.retrieve(organization.stripe_id)
-        if s_customer.email != organization.email:
-            s_customer.email = organization.email
-            s_customer.save()
-    except stripe.error.StripeError:
-        log.exception('Unable to update the Organization billing email on Stripe.')
-    except Exception:
-        log.exception('Unknown error when updating Organization billing email on Stripe.')
+
+    stripe_customer = organization.stripe_customer
+    if not stripe_customer:
+        return
+
+    fields_to_update = {}
+    if organization.email != stripe_customer.email:
+        fields_to_update["email"] = organization.email
+
+    org_metadata = organization.get_stripe_metadata()
+    current_metadata = stripe_customer.metadata or {}
+    for key, value in org_metadata.items():
+        if current_metadata.get(key) != value:
+            current_metadata.update(org_metadata)
+            fields_to_update["metadata"] = current_metadata
```

Comment:
we could also just directly replace it with the current metadata instead of updating, not sure if we have metadata that was set manually or something.

Reply:
Merging the metadata sounds good to me.